From 8e93f6855143a463dceb10ec08d89a734e4e7a7c Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Fri, 18 Mar 2016 14:24:53 -0400
Subject: [PATCH] Fix Furnace cook time bug

If the server lags out and skips multiple ticks, Furnace cooking behavior would not
cook in the expected amount of time as the cook time was not decremented correctly.

This patch ensures that furnaces cook to the correct wall time expectation.

diff --git a/src/main/java/net/minecraft/server/TileEntityFurnace.java b/src/main/java/net/minecraft/server/TileEntityFurnace.java
index 0e3f35a6..37a43fdc 100644
--- a/src/main/java/net/minecraft/server/TileEntityFurnace.java
+++ b/src/main/java/net/minecraft/server/TileEntityFurnace.java
@@ -187,7 +187,7 @@ public class TileEntityFurnace extends TileEntityContainer implements IUpdatePla
         if (this.isBurning() && this.canBurn()) {
             this.cookTime += elapsedTicks;
             if (this.cookTime >= this.cookTimeTotal) {
-                this.cookTime = 0;
+                this.cookTime -= this.cookTimeTotal; // Paper
                 this.cookTimeTotal = this.a(this.items[0]);
                 this.burn();
                 flag1 = true;
-- 
2.50.1

